src/main/java/mate/academy/spring/boot/service/BookServiceImpl.java

```diff
@@ -1,36 +1,48 @@
 package mate.academy.spring.boot.service;
 
-import java.util.Collections;
 import java.util.List;
 import mate.academy.spring.boot.dto.BookDto;
 import mate.academy.spring.boot.dto.BookSearchParametersDto;
 import mate.academy.spring.boot.dto.CreateBookRequestDto;
+import mate.academy.spring.boot.mapper.BookMapper;
 import mate.academy.spring.boot.model.Book;
 import mate.academy.spring.boot.repository.BookRepository;
+import mate.academy.spring.boot.specification.tools.book.spec.builder.impl.BookSpecificationBuilder;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final BookSpecificationBuilder bookSpecificationBuilder;
 
     @Autowired
```

Comment:
Don't need to use @Autowired if you have only one constructor. 
Use `@RequiredArgsConstructor`